lib/src/util/node_modules_importer.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:path/path.dart' as p;
+import 'package:sass/sass.dart';
+
+import '../io.dart';
+
+/// An importer that resolves URLs starting with `~` by searching in
+/// `node_modules` directories.
+///
+/// This doesn't completely match Webpack's behavior (as it lacks the support
+/// for configuring its behavior), but it should be good enough to run the
+/// migrator with for most cases.
+class NodeModulesImporter extends Importer {
+  final _nodeModulesDirectories = <String>[];
+
+  /// Constructs a new [NodeModulesImporter] that searches for `node_modules` in
+  /// [baseDirectory] and all of its ancestors.
+  ///
+  /// If not provided, [baseDirectory] defaults to the current directory.
+  NodeModulesImporter([Directory baseDirectory]) {
```

Comment:
```suggestion
  NodeModulesImporter([String baseDirectory]) {
```

Different codebases differ on this, but I strongly prefer passing around strings for paths and only creating `File` and `Directory` objects at the last possible moment. The only state these classes have is the path anyway, and passing it as a string requires less wrapping when loading from user input and less unwrapping when passing through APIs like the `path` package.